### PR TITLE
pyopenssl 24.3.0. Updated cryptography requirements

### PR DESCRIPTION
**Pull request by timostrunk**
Same as #39, but with updated cryptography dependency.

Closes #39 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by timostrunk** [comment]
I included the hints by the linter.

**Comment by timostrunk** [comment]
Another try to make the linter happy

**Comment by IzerOnadimQC** [comment]
Is this blocked by something or can it be merged?

**Comment by IzerOnadimQC** [comment]
@goanpeca @nehaljwani could one of you merge this? Currently many solves are ending up in a broken state because the `<44` pin for `cryptography` is causing `pyopenssl` to be downgraded to less than `22.0.0`, which is not compatible with later versions of `cryptography`.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-  url: https://pypi.io/packages/source/p/pyOpenSSL/pyopenssl-{{ version }}.tar.gz
-  sha256: 4247f0dbe3748d560dcbb2ff3ea01af0f9a1a001ef5f7c4c647956ed8cbf0e95
+  url: https://pypi.org/packages/source/p/pyOpenSSL/pyopenssl-{{ version }}.tar.gz
+  sha256: 49f7a019577d834746bc55c5fce6ecbcec0f2b4ec5ce1cf43a9a173b8138bb36
 
 build:
   noarch: python
-  number: 2
+  number: 0
   script: {{ PYTHON }} -m pip install . -vvv
```
Comment (ocefpaf) [inline recipe/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

Comment (timostrunk) [reply recipe/meta.yaml]: Thanks! commited

**Comment by jakirkham** [comment]
Thanks Tim and Filipe! 🙏

Perhaps we should consider adding `pip check` and Grayskull: https://github.com/conda-forge/pyopenssl-feedstock/issues/41

**Comment by sfc-gh-wzhao** [comment]
Hi @timostrunk @ocefpaf , looks like the new version has not appeared on conda-forge yet though this PR has been merged for over 5 days, could you please what is happening here? Thank you!

**Comment by jakirkham** [comment]
Thanks for letting us know! 🙏

Looks like CI did not start on `main` for some reason. Have [manually started CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1104132&view=results )

**Comment by jakirkham** [comment]
Looks like the new package is up and am able to download it personally

If you are still seeing issues, please let us know

**Pull request merged**